contrib/platform/src/com/sun/jna/platform/win32/WinBase.java

```diff
@@ -1226,6 +1226,14 @@ public TIME_ZONE_INFORMATION() {
          * Architecture-dependent processor revision.
          */
         public WORD wProcessorRevision;
+
+        // the dwOemID union member is obsolete. Force read of pi instead.
+        @Override
+        public void read() {
+            super.read();
+            processorArchitecture.setType(PI.class);
+            processorArchitecture.read();
```

Comment:
Wouldn't it make more sense to move this to the union itself? I would there override `read()` like this:

```java
public void read() {
    setType(PI.class);
    super.read();
}
```

That way only the correct member will be read (and one unused read is not done).

Reply:
I had that thought initially (and even started with a test case assuming I would do that) but then realized that since this bug(?) has been around for a while, some users may have chosen to work around it by reading the `dwOemID` member.  Changing the behavior might (quietly, without even a deprecation warning) break applications that rely on the current behavior.

Reply:
Ok, so you want to initialize both fields of the union - ok, that should work. I would move it to the union none the less, to keep the functionality together. 

```java
public void read() {
    // Initialize both fields of the union - dwOemID is deprecated, but was by
    // default initialized. To be compatible both fields are initialized.
    setType("dwOemID");
    super.read();
    setType("pi");
    super.read();
}
```